src/charm.py

```diff
@@ -140,36 +141,35 @@ def _check_leader(self):
             self.logger.info("Not a leader, skipping setup")
             raise ErrorWithStatus("Waiting for leadership", WaitingStatus)
 
-    def _deploy_k8s_resources(self) -> None:
-        """Deploys K8S resources."""
+    def _deploy_k8s_resources(self, force: bool = False) -> None:
+        """Deploys K8S resources.
+
+        Args:
+            force (bool): *(optional)* Will "force" apply requests causing conflicting fields to
+                          change ownership to the field manager used in this charm.
+        """
         try:
             self.unit.status = MaintenanceStatus("Creating K8S resources")
-            self.k8s_resource_handler.apply()
-            self.crd_resource_handler.apply()
+            # apply CRDs first
```

Comment:
I think L153-155 deserve a little bit more thought. Since `force` will remove and re-create the resource (causing a service outage) according to the docs [here](https://kubernetes.io/docs/concepts/cluster-administration/manage-deployment/#disruptive-updates) and [here](https://kubernetes.io/docs/reference/kubectl/cheatsheet/#updating-resources), I'd add a check and log what is about to happen. I'd do the following:

```python
try:
  self.k8s_resource_handler.apply()
  self.crd_resource_handler.apply()
except ApiError as api_error:
  if api_error.code == 409:
    self.unit.status = MaintenanceStatus(f"force applying {resource}") # provided you can check which resource is causing the issue
    logger.warning(f"Resource: {resource} will be removed and re-created, you might experience intermittent issues with x and y. If after the refresh process the error persists, please re-apply the removed resource:")
    # here I'd add the yaml of the file, I don't know what would be the best way to do it, 
    # but I'd have some backup of the resource that's going to be removed and re-created
    # naive approach: just log it; other approach iff possible: generate a tmp file with the yaml
    # another approach: show the diff in the logs
    self.k8s_resource_handler.apply(force=True)
  else:
    raise GenericCharmRuntimeError("K8S resource apply failed") from api_error
...
```

Reply:
Tested: resource is updated without delete/create.